lib/authentication/authorize.js

```diff
@@ -36,10 +36,34 @@ module.exports = (options = { models: {} }) => {
       }
 
       try {
-        const wallet = await Wallet.findOne({ userId: req.userData.id });
-        if (!wallet) {
-          logger.warn({ userId: req.userData.id }, 'Wallet record not found');
-          return next(boom.unauthorized());
+        let wallet;
+        if (req.walletData) {
+          /**
+           * If walletData was set by a previous middleware,
+           * search whether Wallet exists for the userId and walletId provided
+           */
+          wallet = await Wallet.findOne({
+            userId: req.userData.id,
+            id: req.walletData.id,
+          });
+          if (!wallet) {
+            logger.warn(
+              { userId: req.userData.id, walletId: req.walletData.id },
+              'Wallet record not found',
+            );
+            return next(boom.unauthorized());
+          }
+        } else {
+          /**
+           * Otherwise, find first wallet created
+           */
+          wallet = await Wallet.findOne({ userId: req.userData.id }).sort({
```

Comment:
You can use a method defined in the schema. `findFirstCreated`

Reply:
It's not using common-models.

Reply:
We are using it when we call the method

const { Wallet } = require('@pillarwallet/common-models').platform;
const authorize = commonMiddleware.authorize({
  logger,
  models: {
    Wallet,
  },
});